DisjointGridSet.java

```diff
@@ -0,0 +1,57 @@
+package assignment4;
+
+import java.util.HashSet;
+import java.util.Set;
+
+
+public class DisjointGridSet {
+
+    private int[] parents;
+    private int initialColumnCount;
+
+    public DisjointGridSet(boolean[][] grid) {
+        this.initialColumnCount = grid[0].length;
+        this.parents = new int[grid.length * grid[0].length];
+        for (int i = 0; i < grid.length; i++) {
+            for (int j = 0; j < grid[0].length; j++) {
+                // initially every island cell is marked as a separate island
+                // -1 for water
+                this.parents[convertIndexes(i, j)] = grid[i][j] ? convertIndexes(i, j) : -1;
+            }
+        }
+    }
+
+    public int find(int element) {
+        if (element == parents[element]) {
+            return element;
+        } else {
+            parents[element] = find(parents[element]);
+            return parents[element];
+        }
+    }
+
+    public void union(int element1, int element2) {

```

Comment:
It would be good to leave notes on Algorithmic time complexity in your code. Then I understand it when reading, and also know that you have thought about it. So I might add javadoc saying:

find = O(n) [as it does a linked list style traversal down the parents array.
convertIndices = O(1)
union = O(n) as it does 2 finds.